public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+    use `JobBoardWithContent`. Of course, we set up the routing configuration for this scenario path accordingly.
+:marked
+  Here, we imitate that hero information is coming from our Hero CRM system, and each hero is represented by a `hero-crm` entity. Just for the sake of
+  demonstration, we add two `hero-crm` tags deliberately with markup. `HeroCrm` could be a standard component, but we implement it as a directive &mdash; 
+  a component without a view and internal logic &mdash; it only holds data:
+  
++makeExample('component-communication/ts/app/take-job-contentchildren/hero-crm.ts', null, 'take-job-contentchildren/hero-crm.ts')
+
+:marked
+  Both the `id` and `name` property is decorated with `@Input()` so that we can use input data binding in `JobBoardWithContent` &mdash; 
+  as we already learned it in the first scenario.
+  
+  ### Adding `@ContentChildren`
+
+  Now, our task is to inject the list of heroes into `HeroJobBoard` so that we can create the corresponding `HeroPanel` components. We are going to use
+  the `@ContentChildren` annotation of Angular. We decorate a component property with `@ContentChildren(C)`, and set the type of the property as `QueryList<C>`,
+  where `C` represents the type of the children. Here is how we use this decorator with `HeroJobBoard`:
+  
++makeExample('component-communication/ts/fragments/heroes-to-invite-2.ts', null, 'take-job-contentchildren/hero-job-board.ts (extract)', {blk: /(@ContentChildren\(HeroCrm\)|QueryList&lt;HeroCrm&gt;)/g, pnk: /(heroesToInvite)/g})
+
+:marked
+  When `HeroJobBoard` is instantiated, the framework assigns a `QueryList<HeroCrm>` instance to `heroesToInvite`. At this moment this list is empty.
+  `QueryList` observes the changes of the parent component's content, updates its content accordingly. Changes in `QueryList`'s content are observed
+  by change detection, and trigger re-rendering, if necessary.
+  
+  When the **Invite heroes** button is clicked, `inviteHeroes()` copies the content of `heroesToInvite` into `heroesInvited`.
+  
+.alert.is-helpful
+  :marked
+    The `inviteHeros()` method uses the `toArray()` method of `QueryList` to convert the `QueryList<HeroCrm>` object to a `Hero[]`. Because the shape of
+    `HeroCrm` is compatible with `Hero` &mdash; they have the same properties &mdash; TypeScript allows this implicit conversion.
+    
+    The component lifecycle has an explicit point when the content of `HeroJobBoard` is collected by the framework, and then `heroesToInvite` is updated. You can hook into
+    this lifecycle event by defining the `ngAfterContentInit()` method, and carry out any additional setup you may need to use the collection of children.
+
+:marked
+  We do not need to change any other components to complete our task. When the app is running, and the **Invite heroes** button is clicked, we 
```

Comment:
And that is part of the problem with this sample.

It doesn't demonstrate that the getting the children has anything at all to do with component communication.  You've shown how to get the children. But you're not using the children in a way that demonstrates how they offer a communication mechanism that's different than anything else you've shown.

You're using the same old binding techniques to the `HeroPanel`. The only difference is that you've found a complex and improbably way to get the heroes via shadow DOM.